### ISSUE TITLE
`declaration-block-trailing-semicolon: never` throws over a bodiless at-rule closing a block with no semicolon, whose end PostCSS leaves unset

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#630]
### What minimal example or steps are needed to reproduce the bug?

```css
a { @v: }
```

or `a { @v }`: a bodiless at-rule closing a block, with no semicolon behind it. PostCSS ends such an at-rule on the block's brace and leaves its `source.end` unset — `postcss.parse("a { @v: }").first.first.source.end` is `undefined` — and `offsetsOf` in `lib/rules/declaration-block-trailing-semicolon/index.ts` throws `The node must carry a source with both of its ends` when `trailingSemicolonIndex` asks where the block's semicolon stands. The whole lint dies, with and without `--fix`.

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/declaration-block-trailing-semicolon": "never"
	}
}
```

`always` does not throw: it asks the block's flag alone and reports the missing semicolon. `at-rule-semicolon-space-before: never` reads the same node and does not throw either.

### What did you expect to happen?

No warning, since the block ends on no semicolon, and the lint to go on.

### What actually happened?

```
Error: The node must carry a source with both of its ends
```

### Notes

Found by the review of #536's pull request, whose race put a Less-shaped variable into a plain CSS file; the throw is the same on `main` at `d726003` and on the branch, so it is nothing #536 made. `offsetsOf` is written for nodes the parser built with both ends, and a bodiless at-rule the container's brace closes is the one node PostCSS hands over without one: the end of such a node is the end of its text, which `raws.between` carries.
